tests/test_functions.py

```diff
@@ -0,0 +1,119 @@
+import requests, difflib, sys
+import re
+
+command_line_arguments = []
```

Comment:
I would just add defaulted parameters for functions where necessary instead of declaring a global variable. 

def function(arg, arg, reset=False)

Reply:
Won't need this global in the future.